lib/resty/websocket/client.lua

```diff
@@ -98,9 +98,17 @@ function _M.connect(self, uri, opts)
         path = "/"
     end
 
-    local ssl_verify, proto_header, origin_header, sock_opts = false
+    local ssl_verify, proto_header, origin_header, custom_header, sock_opts = false
 
     if opts then
+        local headers = opts.headers
+        if headers then
+            custom_header = ""
+            for field,value in pairs(headers) do
+                custom_header = custom_header .. "\r\n" .. field .. ": " .. value
```

Comment:
I have no intention to do anything futher, I will leave using my fork as in my case the performance difference is negligible.